File: group_anagrams.go
func SortString(str string) string {
    s := strings.Split(str, "")
    sort.Strings(s)
    return strings.Join(s, "")
}

func groupAnagrams(strs []string) [][]string {
    mp := make(map[string][]string)
    var ans [][]string

    for _, str := range strs {
        curr := SortString(str)

        mp[curr] = append(mp[curr], str)
    }
    
    for _, val := range mp {
        ans = append(ans, val)
    }
    return ans
}

// Time Complexity : O(n*log(m))
// Space Complexity : O(n)

File: group_anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> mp;
        vector<vector<string>> ans;

        for(auto str: strs) {
            string curr = str;
            sort(curr.begin(), curr.end());

            mp[curr].push_back(str);
        }

        for(auto el: mp) {
            ans.push_back(el.second);
        }
        return ans;
    }
};

// Time Complexity : O(n*log(m))
// Space Complexity : O(n)